### PR TITLE
Fix repetitive api calls and UI issues in connections page 

### LINKED ISSUE #8763: [UI] WASM Filter Import Modal changing focus
### Current Behavior
I can’t type a new name.
I can select a file to upload, but it never appears as a selected file.
As soon as I start to type a new config, the focus jumps somewhere else.

### Environment

- Browser: Safari
- Host OS: Mac 
---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR fixes #8763 
The repetitive api calls in connections page when clicked anywhere in the page and ui issues 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudhanshutech** [comment]
@Yashsharma1911

File: ui/components/connections/index.js [hunk 1]
```diff
@@ -498,7 +498,7 @@ function Connections({
         method : "GET",
       },
       (res) => {
-        setConnections(res?.connections);
+        setConnections(res?.connections || []);
```
Comment (sudhanshutech) [inline ui/components/connections/index.js:501]: in case there is no response

**Comment by Yashsharma1911** [comment]
Thankyou Sudhanhu, can you confirm this fixes the losing focus issue in RJSF modals because of page rerendering which was happing because of listener which listens to toolbar search

**Comment by sudhanshutech** [comment]
> Thankyou Sudhanhu, can you confirm this fixes the losing focus issue in RJSF modals because of page rerendering which was happing because of listener which listens to toolbar search

Yes it does

https://github.com/meshery/meshery/assets/90546692/72bf40ef-62e7-42f5-90a8-cddcc9c54ebc

**Comment by Yashsharma1911** [comment]
> > Thankyou Sudhanhu, can you confirm this fixes the losing focus issue in RJSF modals because of page rerendering which was happing because of listener which listens to toolbar search
> 
> Yes it does
> 
>  Recording.2023-09-19.164807.mp4

Thanks for confirming let's merge it then

**Pull request merged**